content/PGP/YubiKey 5.2.3 Enhancements to OpenPGP 3.4.adoc

```diff
@@ -0,0 +1,168 @@
+
+YubiKeys support the following Elliptic Curve algorithms in addition to RSA (Firmware 5.2.3 and above only)
+
+* secp256r1 
+* secp256k1 
+* secp384r1 
+* secp521r1 
+* brainpoolP256r1 
+* brainpoolP384r1 
+* brainpoolP512r1 
+* curve25519 
+** x25519 (decipher only) 
+** ed25519 (sign / auth only) 
+
+=== Sample usage [GnuPG (2.2.12+)]:
+
+* curve25519 or secp384r1
+[source, python]
+-----------------
+$ gpg --edit-card
+gpg/card> admin
+gpg/card> key-attr
+Changing card key attribute for: Signature key
+Please select what kind of key you want:
+  (1) RSA
+  (2) ECC
+Your selection? 2
+Please select which elliptic curve you want:
+  (1) Curve 25519
+  (4) NIST P-384
+Your selection? 1
+gpg/card> generate
+-----------------
+* all curves
+[source, python]
+-----------------
+$ gpg --edit-card --expert 
+-----------------
+* Set slot attributes with scdaemon
+[source, python]
+-----------------
+$ gpg-connect-agent "SCD SETATTR KEY-ATTR --force <SLOT> <ALGO ID> <ALGO>" /bye 
+-----------------
+* RSA
+[source, python]
+-----------------
+$ gpg-connect-agent "SCD SETATTR KEY-ATTR --force 1 1 rsa2048" /bye 
+-----------------
+* ECDSA
+[source, python]
+-----------------
+$ gpg-connect-agent "SCD SETATTR KEY-ATTR --force 1 19 nistp256" /bye
+$ gpg-connect-agent "SCD SETATTR KEY-ATTR --force 1 19 brainpoolP256r1" /bye
+-----------------
+* curve25519
+[source, python]
+-----------------
+$ gpg-connect-agent "SCD SETATTR KEY-ATTR --force 1 22 ed25519" /bye
+$ gpg-connect-agent "SCD SETATTR KEY-ATTR --force 2 22 cv25519" /bye
+-----------------
+
+=== Attestation
+YubiKeys offer a new feature to the OpenPGP Smart Card, the attestation of Keys generated on device. With the attestation function, generating an Authentication, Signature or Decipher key on a YubiKey will also create an X.509 certificate signed with the Attestation key present on the device. This certificate can be used to verify any other key generated on the device and was not imported, ensuring said key only exists on the YubiKey. 
+
+Attestation is supported for all combinations of attestation key and attested key, except the attestation key cannot be from the curve25519 family. A fourth key slot is reserved for an attestation key.  If specifying it by ID, it is Key ID 0x81. The Attestation template certificate Is stored on a fourth user certificate slot reserved for it. It is selected directly with SELECT_DATA index 4.  It can also be retrieved by calling GET_NEXT_DATA three times after SELECT_DATA on index 1.
+
+Each YubiKey comes with a pre-loaded Attestation certificate signed by a Yubico OpenPGP CA. Attestation is a Yubikey-proprietary extension. The https://www.google.com/url?q=https://www.yubico.com/products/services-software/download/yubikey-manager/&sa=D&ust=1566260477099000[YubiKey Manager] application can be used to manage the Attestation Certificate. For more details, refer to the OpenPGP Attestation page at: https://developers.yubico.com/PGP/Attestation.html

```

Comment:
I think the google part can be removed in that URL: https://www.google.com/url?q=